### PR TITLE
Do not show SVN warning messages

### PR DESCRIPTION
**Pull request by jgsogo**
Changelog: omit
Docs: omit

**Wait for https://github.com/conan-io/conan/pull/4890 ⚠️ ⚠️ ⚠️** 

@PYVERS: Linux@py37
@tags: svn

----

In our test suite, there are a lot of messages like the following ones that are printed each time SVN is called:

```
svn: warning: cannot set LC_CTYPE locale
svn: warning: environment variable LC_ALL is en_US.UTF-8
svn: warning: please check that your locale name is correct
```

This message appears because the locale `en_US.UTF-8` is not installed in the Jenkins machines and it is set before each call to the SCM console ([here](https://github.com/conan-io/conan/blob/develop/conans/client/tools/scm.py#L54)):

```bash
>locale -a
C
C.UTF-8
POSIX
```

Not sure, why we were setting the `LC_ALL` for the Git calls (if _force english_), and only for this system call (we are doing nothing similar for other tools we are calling through the console).

Two different options have succeeded in this PR, but there are other alternatives:
 * can we think about removing this statement that adds the `LC_ALL` env variable? _<-- Jenkins green as in ca83cff_
 * can we install those locales?
 * can we detect which locales are installed and use one of them?
 * apply this locale only for Git
 * ... or just throw away `stderr` from SVN (pipe, or null) _<-- Jenkins green e02201a_

WDYT?

### CODE REVIEW

**Comment by lasote** [comment]
About why we set it, to force the output to be English. I think this was raised by aalman (Germany) in this feature beginnings.
But that is a different discussion. The purpose of this, no matter what generate them, are those warnings  printed in the console escaping from the conan buffering. So only  e02201a_ is the solution, the rest is not relevant and also I think they are correct.

File: conans/client/tools/scm.py [hunk 1]
```diff
@@ -218,7 +218,7 @@ class SVN(SCMBase):
 
     def __init__(self, folder=None, runner=None, *args, **kwargs):
         def runner_no_strip(command):
-            return decode_text(subprocess.check_output(command, shell=True))
+            return decode_text(subprocess.check_output(command, shell=True, stderr=PIPE))
```
Comment (lasote) [inline conans/client/tools/scm.py]: Shouldn't be `stderr=STDOUT`?

Comment (lasote) [reply conans/client/tools/scm.py]: You are right, if we `strerr=STDOUT` we can get errors when we expect stdout, not a good idea. Probably the origin of #4850

**Comment by jgsogo** [comment]
This could go to v1.14 like https://github.com/conan-io/conan/pull/4853

**Comment by lasote** [comment]
Should this still be open?

**Comment by jgsogo** [comment]
Probably not, but let's wait for the `check_output` one to be merged and see if there is any diff to take into account.

**Comment by lasote** [comment]
Update with develop

**Comment by jgsogo** [comment]
No changes :+1:

**Pull request closed without merging**